catalog_impls/src/volatile.rs

```diff
@@ -17,51 +16,82 @@ use catalog::{
     schema::{
         self, CatalogMismatch, CloseOptions, CloseTable, CloseTableRequest, CreateOptions,
         CreateTable, CreateTableRequest, DropOptions, DropTable, DropTableRequest, NameRef,
-        OpenOptions, OpenTable, OpenTableRequest, Schema, SchemaMismatch, SchemaRef,
+        OpenOptions, OpenTable, OpenTableRequest, Schema, SchemaMismatch, SchemaName, SchemaRef,
     },
     Catalog, CatalogRef,
 };
-use log::info;
+use log::{debug, info};
 use snafu::{ensure, ResultExt};
 use table_engine::table::{SchemaId, TableId, TableRef};
 use tokio::sync::Mutex;
 
 #[async_trait]
-pub trait SchemaIdGenerator {
+pub trait SchemaIdAlloc: Send + Sync {
     type Error: std::error::Error + Send + Sync + 'static;
     async fn alloc_schema_id(
         &self,
-        schema_name: NameRef,
+        schema_name: &str,
     ) -> std::result::Result<SchemaId, Self::Error>;
 }
 
 #[async_trait]
-pub trait TableIdGenerator {
+pub trait TableIdAlloc: Send + Sync {
     type Error: std::error::Error + Send + Sync + 'static;
     async fn alloc_table_id(
         &self,
-        schema_id: SchemaId,
-        table_name: NameRef,
+        schema_name: &str,
+        table_name: &str,
     ) -> std::result::Result<TableId, Self::Error>;
 
     async fn invalidate_table_id(
         &self,
-        schema_id: SchemaId,
+        schema_name: &str,
```

Comment:
Why we deprecate schema id here

Reply:
Actually, both id and name are ok but it is more convenient for implementing this trait based on meta client.